### PR TITLE
Feature/meraki switch

### PR DESCRIPTION
**Pull request by whitx**
# Description

Allow the use of the switch MS220 and MS220-8P
# Impacts

Meraki switch module
# Delete branch after merge

NO 
# NEWS file entries

Meraki switch MS220_8 and MS200_8P are now supported into PacketFence, with Vlan enforcement support, using MAB or dot1x.
## Enhancements
- Meraki switch module

### CODE REVIEW

**Comment by extrafu** [comment]
Provide configuration documentation for the network conf guide.

**Comment by whitx** [comment]
Working on WebAuth support and doc this is also why there is a 'not ready'!

File: lib/pf/Switch/Meraki/MS220_8.pm [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+client mac address
+SSID
+client ip address
+redirect url
+grant url
+status code
+
+=cut
+
+sub parseUrl {
```
Comment (julsemaan) [inline lib/pf/Switch/Meraki/MS220_8.pm]: If it only supports VLAN assignment as you state above, why do you put this here

File: lib/pf/Switch/Meraki/MS220_8.pm [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    my($self, $req, $r) = @_;
+    my $logger = $self->logger;
+    
+    return ($$req->param('client_mac'),$$req->param('wlan'),$$req->param('client_ip'),$$req->param('redirect'),$$req->param('switch_url'),$$req->param('statusCode'));
+
+}
+
+sub parseSwitchIdFromRequest {
```
Comment (julsemaan) [inline lib/pf/Switch/Meraki/MS220_8.pm]: Same for this

File: lib/pf/Switch/Meraki/MS220_8.pm [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+=cut
+
+# CAPABILITIES
+# access technology supported
+sub description { 'Meraki switch MS220_8' }
+sub supportsWiredMacAuth { return $TRUE; }
+sub supportsWiredDot1x { return $TRUE; }
+sub supportsExternalPortal { return $FALSE; }
+sub supportsWebFormRegistration { return $FALSE }
```
Comment (julsemaan) [inline lib/pf/Switch/Meraki/MS220_8.pm]: This is already false in pf::Switch, remove

File: lib/pf/Switch/Meraki/MS220_8.pm [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+=head1 SUBROUTINES
+
+=cut
+
+# CAPABILITIES
+# access technology supported
+sub description { 'Meraki switch MS220_8' }
+sub supportsWiredMacAuth { return $TRUE; }
+sub supportsWiredDot1x { return $TRUE; }
+sub supportsExternalPortal { return $FALSE; }
```
Comment (julsemaan) [inline lib/pf/Switch/Meraki/MS220_8.pm]: Remove, its the same as pf::Switch

**Comment by whitx** [comment]
UP for review, tested MAB/dot1x

File: lib/pf/Switch/Meraki/MS220_8.pm [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+The pf::Switch::Meraki::MS220_8 module implements an object oriented interface to
+manage the connection with MS220_8 switch model.
+
+=head1 STATUS
+
+Developed and tested on a MS220_8P (P standing for PoE) switch
+
+=head1 BUGS AND LIMITATIONS
+
+The firmware allow only for VLAN enforcement at the moment. We cannot push the predfine policies from PacketFence.
```
Comment (fdurand) [inline lib/pf/Switch/Meraki/MS220_8.pm]: predefined

**Comment by fdurand** [comment]
Looks good, just one thing is missing in my opinion, this is the Meraki module (something like pf::Switch::Netgear) and in MS220_8.pm use base ('pf::Switch:Meraki');

**Comment by julsemaan** [comment]
Hum this doesn't look good at all, you're merging wayyyy too much stuff. You probably f****d up something while playing with git

**Comment by whitx** [comment]
Yeah working on it

**Comment by whitx** [comment]
Rebased and ready

File: lib/pf/Switch/Meraki/MS220_8.pm [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+use strict;
+use warnings;
+
+use base ('pf::Switch::Meraki');
+
+use pf::constants;
+use pf::config;
+use pf::util;
+use pf::node;
```
Comment (fdurand) [inline lib/pf/Switch/Meraki/MS220_8.pm]: Useless

File: lib/pf/Switch/Meraki/MS220_8.pm [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+=cut
+
+use strict;
+use warnings;
+
+use base ('pf::Switch::Meraki');
+
+use pf::constants;
+use pf::config;
```
Comment (fdurand) [inline lib/pf/Switch/Meraki/MS220_8.pm]: Useless

**Pull request merged**